Packs/FeedThreatFox/Integrations/FeedThreatFox/FeedThreatFox.py

```diff
@@ -0,0 +1,405 @@
+import pytz
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+from CommonServerUserPython import *  # noqa
+
+import urllib3
+from typing import Any
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+LOG_LINE = "THREAT FOX-"
+
+
+class Client(BaseClient):
+    def get_indicators_request(self, query: dict) -> dict:
+        url_suffix = '/api/v1'
+        body = query
+        return self._http_request('POST', url_suffix=url_suffix, json_data=body)
+
+    def test_module(self) -> str:
+        """Tests API connectivity and authentication'
+
+        Returning 'ok' indicates that the integration works like it is supposed to.
+        Connection to the service is successful.
+        Raises exceptions if something goes wrong.
+
+        :type client: ``Client``
+        :param Client: client to use
+
+        :return: 'ok' if test passed, anything else will fail the test.
+        :rtype: ``str``
+        """
+        self.get_indicators_request({'days': 1, 'limit': 5})
+        return 'ok'
+
+
+def check_args_for_query(args: dict) -> tuple[bool, str | None]:
+    """Checks that the args given are valid.
+       Valid args are: Exactly one field to search by (one of these:
+       'search_term', 'id', 'hash', 'tag', 'malware')
+    Args:
+        args: dict
+    Returns:
+        Boolean: True if params are good and False otherwise.
+        Str: The query type (one of these: 'search_term', 'id', 'hash', 'tag', 'malware').
+            If args are not good than it will be None.
+    """
+    args_lst = list(args.keys())
+    if 'limit' in args_lst:
+        args_lst.remove('limit')
+    if len(args_lst) != 1:
+        raise DemistoException("Arguments given are invalid. Please specify exactly one argument to search by.")
+    else:
+        return args_lst[0]
+
+
+def create_query(query_arg, id: str | None = None, search_term: str | None = None,
+                 hash: str | None = None, tag: str | None = None, malware: str | None = None,
+                 days: str | None = None, limit: str | None = None) -> dict:
+    """Creates a valid query to send to the API.
+
+    Args:
+        query_arg (str): the query type (should be one of those:
+        'search_term', 'id', 'hash', 'tag', 'malware', 'days').
+
+    Returns:
+        Str: The query to send to the API.
+    """
+
+    query_dict = {'search_term': 'search_ioc', 'id': 'ioc', 'hash': 'search_hash',
+                  'tag': 'taginfo', 'malware': 'malwareinfo'}
+
+    q_id = arg_to_number(id)
+
+    query = assign_params(
+        query=query_dict[query_arg],
+        id=q_id,
+        search_term=search_term,
+        hash=hash,
+        tag=tag,
+        malware=malware,
+    )
+
+    # Only queries searching by tag or malware can specify a limit.
+    if query_arg == 'tag' or query_arg == 'malware':
```

Comment:
Consider changing to a list and check if the value is in it for readability, aka LIMIT_SUPPORTED_QUERIES = ['tags', 'malware'].